app/services/AuditService.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import com.google.inject.Inject
+import config.AppConfig
+import models.{ContactDetails, FileUpload, FileUploadCount, MRN, MessageFilterTag}
+import play.api.Logging
+import play.api.libs.json.{JsObject, Json}
+import services.AuditTypes.{NavigateToMessages, UploadSuccess}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.{AuditConnector, AuditResult}
+import uk.gov.hmrc.play.audit.AuditExtensions
+import uk.gov.hmrc.play.audit.AuditExtensions._
+import uk.gov.hmrc.play.audit.http.connector.AuditResult.{Disabled, Failure, Success}
+import uk.gov.hmrc.play.audit.model.{DataEvent, ExtendedDataEvent}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AuditService @Inject()(connector: AuditConnector, appConfig: AppConfig)(implicit ec: ExecutionContext) extends Logging {
+
+  def auditSecureMessageInbox(enrolment: String, eori: String, filter: MessageFilterTag, path: String)(
+    implicit hc: HeaderCarrier
+  ): Future[AuditResult] = {
+    val auditType = NavigateToMessages
+    val details = Map("enrolment" -> enrolment, "eoriNumber" -> eori)
+    val tags = Map("notificationType" -> filter.filterValue)
+
+    val extendedEvent = ExtendedDataEvent(
+      auditSource = appConfig.appName,
+      auditType = auditType.toString,
+      detail = Json.toJson(details).as[JsObject] + ("tags" -> Json.toJson(tags)),
```

Comment:
IMO it's a bit convoluted. Why instead to create and merge 2 maps (with an additional cast) not use something simpler like
```
Json.obj(
  "enrolment" -> enrolment,
  "eoriNumber" -> eori,
  "tags" -> Json.obj("notificationType" -> notificationType)
)
```

Reply:
Yes, much nicer. Thanks :+1: